gov/governance/governance_proposal.gno

```diff
@@ -174,6 +174,37 @@ func ProposeParameterChange(
 		panic("PROPOSER_HAS_NOT_ENOUGH_XGNS")
 	}
 
+	if numToExecute == 0 {
+		panic("numToExecute is 0")
+	}
+
+	// check if numToExecute is a valid number
+	splitGov := strings.Split(executions, "*GOV*")
+	if uint64(len(splitGov)) != numToExecute {
+		panic(ufmt.Sprintf("numToExecute(%d) does not match the number of executions(%d)", numToExecute, len(splitGov)))
+	}
+
+	// check if each execution is valid
+	for _, gov := range splitGov {
+		splitExe := strings.Split(gov, "*EXE*")
+
+		var pkgPath, funcName, params string
+		switch len(splitExe) {
+		case 3:
+			pkgPath = splitExe[0]
+			funcName = splitExe[1]
+			params = splitExe[2]
+		case 2:
+			pkgPath = splitExe[0]
+			funcName = splitExe[1]
```

Comment:
It seems like it would be good to manage the extracted results as structures. I'll check on this later